bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/SAICiSMARTHandler.java

```diff
@@ -0,0 +1,278 @@
+import net.heberling.ismart.asn1.v2_1.entity.RvcReqParam;
+import net.heberling.ismart.asn1.v3_0.entity.OTA_ChrgMangDataResp;
+
+/**
+ * The {@link SAICiSMARTHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Heberling - Initial contribution
+ */
+@NonNullByDefault
+public class SAICiSMARTHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SAICiSMARTHandler.class);
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    @Nullable
+    SAICiSMARTVehicleConfiguration config;
+    private @Nullable Future<?> pollingJob;
+    private ZonedDateTime lastAlarmMessage;
+    private ZonedDateTime lastCarActivity;
+
+    /**
+     * If the binding is initialized, treat the car as active (lastCarActivity = now) to get some first data.
+     * 
+     * @param httpClientFactory
+     * @param timeZoneProvider
+     * @param thing
+     */
+    public SAICiSMARTHandler(TimeZoneProvider timeZoneProvider, Thing thing) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+        lastAlarmMessage = ZonedDateTime.now(getTimeZone());
+        lastCarActivity = ZonedDateTime.now(getTimeZone());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(SAICiSMARTBindingConstants.CHANNEL_FORCE_REFRESH) && command == OnOffType.ON) {
+            // reset channel to off
+            updateState(CHANNEL_FORCE_REFRESH, OnOffType.from(false));
+            // update internal activity date, to query the car for about a minute
+            notifyCarActivity(ZonedDateTime.now(getTimeZone()).minus(9, ChronoUnit.MINUTES), true);
+        } else if (channelUID.getId().equals(CHANNEL_SWITCH_AC) && command == OnOffType.ON) {
+            // reset channel to off
+            updateState(CHANNEL_SWITCH_AC, OnOffType.ON);
+            // enable air conditioning
+            try {
+                sendACCommand((byte) 5, (byte) 8);
+            } catch (URISyntaxException | ExecutionException | TimeoutException | InterruptedException e) {
+                logger.warn("A/C On Command failed", e);
+            }
+        } else if (channelUID.getId().equals(CHANNEL_SWITCH_AC) && command == OnOffType.OFF) {
+            // reset channel to off
+            updateState(CHANNEL_SWITCH_AC, OnOffType.OFF);
+            // disable air conditioning
+            try {
+                sendACCommand((byte) 0, (byte) 0);
+            } catch (URISyntaxException | ExecutionException | TimeoutException | InterruptedException e) {
+                logger.warn("A/C Off Command failed", e);
+            }
+        } else if (channelUID.getId().equals(CHANNEL_LAST_ACTIVITY) && command instanceof DateTimeType) {
+            // update internal activity date from external date
+            notifyCarActivity(((DateTimeType) command).getZonedDateTime(), true);
```

Comment:
```suggestion
                } else if (channelUID.getId().equals(CHANNEL_LAST_ACTIVITY) && command instanceof DateTimeType                 commnadAsDateTimeType) {
            // update internal activity date from external date
            notifyCarActivity(commnadAsDateTimeType.getZonedDateTime(), true);
```

You can use java 17 pattern matching.
```suggestion

```